src/Subscriber.ts

```diff
@@ -199,80 +191,49 @@ class SafeSubscriber<T> extends Subscriber<T> {
 
   next(value?: T): void {
     if (!this.isStopped && this._next) {
-      const { _parentSubscriber } = this;
-      if (!_parentSubscriber.syncErrorThrowable) {
-        this.__tryOrUnsub(this._next, value);
-      } else if (this.__tryOrSetError(_parentSubscriber, this._next, value)) {
+      try {
+        this._next.call(this._context, value);
+      } catch (err) {
         this.unsubscribe();
+        this._hostReportError(err);
```

Comment:
not sure if this need to be aligned - cases like error / complete we do report error first then unsubscribe, for next we do unsubscribe first then report. Nitpick maybe?

Reply:
The error is reported asynchronously, so in this case it doesn't matter. However, I think it's probably supposed to be report error then unsubscribe. I'll fix it.

Reply:
Improved.